### PR TITLE
Improvement of csproj finding

### PR DESCRIPTION
**Pull request by WojciechNagorski**
During writing the blog post about running BenchmarkDotNet in a Docker container (https://wojciechnagorski.com/2019/12/how-to-run-benchmarkdotnet-in-a-docker-container/) I found some problems with finding a `csproj` file. 

Reproduction steps:
```
dotnet new console --name TestApp

cd TestApp

dotnet add package BenchmarkDotNet

```

The `Program.cs` file:

```
using System;
using BenchmarkDotNet.Attributes;
using BenchmarkDotNet.Running;

namespace TestApp
{
    class Program
    {
        static void Main(string[] args)
        {
            BenchmarkSwitcher.FromAssembly(typeof(Program).Assembly).Run(args);
        }
    }

    public class MyBenchmark
    {
        [Benchmark]
        public int Sum()
        {
            int result = 0;
            for (int i = 0; i < 100; i++)
            {
                result += i;
            }

            return result;
        }
    }
}
```

After you run a benchmark, using the following command, everything will be fine:
```
dotnet run -c Release -- --filter *
```

But if you run a benchmark, with the following instructions, you will get the exception:

```
dotnet build -c Release
cd bin\Release\netcoreapp3.0
dotnet TestApp.dll --filter *
```
The output with exception:
```
// Validating benchmarks:
// ***** BenchmarkRunner: Start   *****
// ***** Found 1 benchmark(s) in total *****
// ***** Building 1 exe(s) in Parallel: Start   *****
Unable to find .sln file. Will use current directory c:\work\TestApp\bin\Release\netcoreapp3.0 to search for project file. If you don't use .sln file on purpose it should not be a problem.
// ***** Done, took 00:00:00 (0.11 sec)   *****
// Found 1 benchmarks:
//   MyBenchmark.Sum: DefaultJob

// Generate Exception: Unable to find TestApp in c:\work\TestApp\bin\Release\netcoreapp3.0 and its subfolders. Most probably the name of output exe is different than the name of the .(c/f)sproj

// ***** BenchmarkRunner: Finish  *****

// * Export *
  BenchmarkDotNet.Artifacts\results\TestApp.MyBenchmark-report.csv
  BenchmarkDotNet.Artifacts\results\TestApp.MyBenchmark-report-github.md
  BenchmarkDotNet.Artifacts\results\TestApp.MyBenchmark-report.html

// * Detailed results *
MyBenchmark.Sum: DefaultJob
Runtime = ; GC =
There are not any results runs

// * Summary *

BenchmarkDotNet=v0.12.0, OS=Windows 10.0.16299.1387 (1709/FallCreatorsUpdate/Redstone3)
Intel Core i7-4770 CPU 3.40GHz (Haswell), 1 CPU, 8 logical and 4 physical cores
Frequency=3312641 Hz, Resolution=301.8739 ns, Timer=TSC
.NET Core SDK=3.0.100
  [Host] : .NET Core 3.0.0 (CoreCLR 4.700.19.46205, CoreFX 4.700.19.46214), X64 RyuJIT


| Method | Mean | Error |
|------- |-----:|------:|
|    Sum |   NA |    NA |

Benchmarks with issues:
  MyBenchmark.Sum: DefaultJob

// * Legends *
  Mean  : Arithmetic mean of all measurements
  Error : Half of 99.9% confidence interval
  1 ns  : 1 Nanosecond (0.000000001 sec)

// ***** BenchmarkRunner: End *****
// ** Remained 0 benchmark(s) to run **
Run time: 00:00:00 (0.23 sec), executed benchmarks: 0

Global total time: 00:00:00 (0.35 sec), executed benchmarks: 0
// * Artifacts cleanup *

c:\work\TestApp\bin\Release\netcoreapp3.0
```

A solution for this problem is creating `sln` file, but I think that BenchmarkDotNet should work without it too.

### CODE REVIEW

File: src/BenchmarkDotNet/Toolchains/CsProj/CsProjGenerator.cs [hunk 1]
```diff
@@ -140,11 +140,14 @@ protected virtual string GetRuntimeSettings(GcMode gcMode, IResolver resolver)
         [PublicAPI]
         protected virtual FileInfo GetProjectFilePath(Type benchmarkTarget, ILogger logger)
         {
-            if (!GetSolutionRootDirectory(out var solutionRootDirectory))
+            if (!GetSolutionRootDirectory(out var rootDirectory))
```
Comment (adamsitnik) [inline src/BenchmarkDotNet/Toolchains/CsProj/CsProjGenerator.cs]: nit: to reduce the nesting, could you please join the both conditions using `&&` ?

```cs
if (!GetSolutionRootDirectory(out var rootDirectory) && !GetProjectRootDirectory(out rootDirectory))
```

File: src/BenchmarkDotNet/Toolchains/DotNetCli/DotNetCliGenerator.cs [hunk 2]
```diff
@@ -88,5 +99,10 @@ private static bool IsRootSolutionFolder(DirectoryInfo directoryInfo)
             => directoryInfo
                 .GetFileSystemInfos()
                 .Any(fileInfo => fileInfo.Extension == ".sln" || fileInfo.Name == "global.json");
+        
+        private static bool IsRootProjectFolder(DirectoryInfo directoryInfo)
+            => directoryInfo
+                .GetFileSystemInfos()
+                .Any(fileInfo => fileInfo.Extension == ".csproj");
```
Comment (adamsitnik) [inline src/BenchmarkDotNet/Toolchains/DotNetCli/DotNetCliGenerator.cs]: damn, I have forgotten about F# and VB projects. Could you please handle `.fsproj` and `.vbroj` files as well?

Comment (adamsitnik) [reply src/BenchmarkDotNet/Toolchains/DotNetCli/DotNetCliGenerator.cs]: @WojciechNagorski ping

**Comment by WojciechNagorski** [comment]
I will fix it. Sorry for late, I was busy.

**Pull request merged**